policy/assertion_test.go

```diff
@@ -132,3 +199,108 @@ func TestNewAssertion(t *testing.T) {
 		})
 	}
 }
+
+func Test_isRegexMetaCharacter(t *testing.T) {
+	type args struct {
+		target rune
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "check '^', return true",
+			args: args{
+				target: '^',
+			},
+			want: true,
+		},
+		{
+			name: "check '$', return true",
+			args: args{
+				target: '$',
+			},
+			want: true,
+		},
+		{
+			name: "check '.', return true",
+			args: args{
+				target: '.',
+			},
+			want: true,
+		},
+		{
+			name: "check ')', return true",
+			args: args{
+				target: ')',
+			},
+			want: true,
+		},
+		{
+			name: "check '*', return false",
+			args: args{
+				target: '*',
+			},
+			want: false,
+		},
+		{
+			name: "check ']', return false",
+			args: args{
+				target: ']',
+			},
+			want: false,
+		},
+		{
+			name: "check 'a', return false",
+			args: args{
+				target: 'a',
+			},
+			want: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := isRegexMetaCharacter(tt.args.target); got != tt.want {
+				t.Errorf("isRegexMetaCharacter() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_patternFromGlob(t *testing.T) {
+	type args struct {
+		glob string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
+			name: "check to escape regular expression meta characters",
+			args: args{
+				glob: "^$.|[+\\(){",
+			},
+			want: "^\\^\\$\\.\\|\\[\\+\\\\\\(\\)\\{$",
+		},
+		{
+			name: "check to convert wildcards",
+			args: args{
+				glob: ".*wild?card?-*test*",
+			},
+			want: "^\\..*wild.card.-.*test.*$",
+		},
```

Comment:
Please also add test case about `"\*"`. To show escape is not supported.

Reply:
Fixed 6c1991f